src/engraving/dom/measure.cpp

```diff
@@ -1534,7 +1534,7 @@ EngravingItem* Measure::drop(EditData& data)
             score()->undoChangeKeySig(staff, tick(), k);
         } else {
             // apply to all staves:
-            for (Staff* s : score()->staves()) {
+            for (Staff* s : score()->masterScore()->staves()) {
                 score()->undoChangeKeySig(s, tick(), k);
```

Comment:
Shouldn't this be changed to `score()->masterScore()` as well? Even when it doesn't cause problems right now, it feels scary to call `score()->undoChangeKeySig` with a staff that doesn't belong to `score()`.

Reply:
Oh you're right, I didn't even notice that. I just copied what was done in KeySig::drop. Honestly that's just another indication that this issue is so old and full of jank that it doesn't make sense to try to tackle it now. I'll just postpone it